sympy/geometry/polygon.py

```diff
@@ -783,6 +783,87 @@ def intersection(self, o):
             return list(ordered(intersection_result))
 
 
+    def cut_section(self, line):
+        """
+        Returns a tuple of two polygon segments that lie above and below
+        the intersecting line respectively.
+
+        References
+        ==========
+
+        https://github.com/sympy/sympy/wiki/A-method-to-return-a-cut-section-of-any-polygon-geometry
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Symbol, Polygon, Line
+        >>> a, b = 20, 10
+        >>> p1, p2, p3, p4 = [(0, b), (0, 0), (a, 0), (a, b)]
+        >>> rectangle = Polygon(p1, p2, p3, p4)
+        >>> t = rectangle.cut_section(Line((0, 5), slope=0))
```

Comment:
The code is giving `ValueError` for the line passing through `(0, 0)` and having `slope=0`
```
>>> a, b = 20, 10
>>> p1, p2, p3, p4 = [(0, b), (0, 0), (a, 0), (a, b)]
>>> Polygon(p1, p2, p3, p4).cut_section(Line((0, 0), slope=0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jashan/sympy/sympy/geometry/polygon.py", line 864, in cut_section
    return Polygon(*upper_vertices), Polygon(*lower_vertices)
  File "/home/jashan/sympy/sympy/geometry/polygon.py", line 156, in __new__
    return Point(*vertices, **kwargs)
  File "/home/jashan/sympy/sympy/geometry/point.py", line 138, in __new__
    keyword `dim` > 1.'''))
ValueError: 
Point requires 2 or more coordinates or keyword `dim` > 1.

```

Reply:
I think that occurs as there is no polygon below the line and hence the list of `lower_vertices` goes empty, so it is not able to make a polygon from it.
Should we be returning `None` in such a case?

Reply:
When the line is parallel to the upper side of the rectangle, it returns `Segment2D(Point2D(0, 10), Point2D(20, 10))`:
```
>>> a, b = 20, 10
>>> p1, p2, p3, p4 = [(0, b), (0, 0), (a, 0), (a, b)]
>>> Polygon(p1, p2, p3, p4).cut_section(Line((0, b), slope=0))
(Segment2D(Point2D(0, 10), Point2D(20, 10)), Polygon(Point2D(0, 10), Point2D(0, 0), 
Point2D(20, 0), Point2D(20, 10)))
```

Reply:
> Should we be returning None in such a case?

It should be fine as long as both of the above cases show similar behavior.